wildfly/connectors/cassandra/connector-cassandra/src/main/java/org/teiid/resource/adapter/cassandra/CassandraConnectionImpl.java

```diff
@@ -18,64 +18,33 @@
 
 package org.teiid.resource.adapter.cassandra;
 
-import java.util.List;
-import java.util.Set;
-
-import javax.resource.ResourceException;
-
+import org.teiid.cassandra.BaseCassandraConnection;
 import org.teiid.logging.LogConstants;
 import org.teiid.logging.LogManager;
-import org.teiid.resource.spi.BasicConnection;
-import org.teiid.translator.TranslatorException;
-import org.teiid.translator.cassandra.CassandraConnection;
-
+import org.teiid.resource.spi.ResourceConnection;
 import com.datastax.driver.core.*;
 
+import javax.resource.ResourceException;
+
 /**
  * Represents a connection to Cassandra database.
  * */
-public class CassandraConnectionImpl extends BasicConnection implements CassandraConnection{
-    private CassandraManagedConnectionFactory config;
-    private Cluster cluster = null;
-    private Session session = null;
-    private Metadata metadata = null;
-    private VersionNumber version;
-
-    public CassandraConnectionImpl(CassandraManagedConnectionFactory config, Metadata metadata) {
-        this.config = config;
-        this.metadata = metadata;
-    }
+public class CassandraConnectionImpl extends BaseCassandraConnection implements ResourceConnection {
 
     public CassandraConnectionImpl(CassandraManagedConnectionFactory config) {
-        this.config = config;
-
-        Cluster.Builder builder  = Cluster.builder().addContactPoint(config.getAddress());
-
-        if (this.config.getUsername() != null) {
-            builder.withCredentials(this.config.getUsername(), this.config.getPassword());
-        }
-
-        if (this.config.getPort() != null) {
-            builder.withPort(this.config.getPort());
-        }
-
-        this.cluster = builder.build();
-
-        this.metadata = cluster.getMetadata();
-
-        this.session = cluster.connect(config.getKeyspace());
+        super(config);
+    }
 
-        Set<Host> allHosts = cluster.getMetadata().getAllHosts();
-        if (!allHosts.isEmpty()) {
-            Host host = allHosts.iterator().next();
-            this.version = host.getCassandraVersion();
-        }
+    public CassandraConnectionImpl(CassandraManagedConnectionFactory config, Metadata metadata) {
+        super(config, metadata);
     }
 
     @Override
     public void close() throws ResourceException {
-        if(cluster != null){
-            cluster.close();
+        try {
+            super.close();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
> Avoid printStackTrace. At first glance you can just throw a ResourceException.
> Ok I get this. I just didn't give this a thought at that time.
> However you can avoid all of this by changing the base close method - does it really need to throw Exception? It looks like it doesn't need any checked exception thrown - which means it won't need any override in extending classes.
I am not sure what you mean here. What I understand is since the close method in baseCassandraConnection need no checked exception thrown, I need not throw any exception from it and thus we won't need to handle any exception in the overridden classes, which here is the CassandraConnectionImpl i.e it won't throw the ResourceException. Right?

Reply:
> it won't throw the ResourceException. Right?

Correct, the better fix is just removing all of the checked exception throwing.

Reply:
I think it's fixed now!